### ISSUE TITLE
Simplify register access in hardware.c

### ISSUE THREAD

**Issue by tkuester** [esp32-open-mac/esp32-open-mac#1]
Hey, congratulations on the project! Very exciting developments for the esp32 community!

I'm not sure if this would be appropriate for the project, but it's a fun little trick from the AVR world that may help simplify some code. (It all compiles out to be the same, I'm sure.)

```c
#define _MMIO_DWORD(mem_addr) (*(volatile uint32_t *)(mem_addr))

#define WIFI_DMA_OUTLINK _MMIO_DWORD(0x3ff73d20)
```

Now you can do bit sets in much simpler (and perhaps more familiar) code.

```c
WIDI_DMA_OUTLINK |= bitmask;
```

**Comment by tristanbruns** [comment]
What you describe is done in code targeting AVR, because the AVR instruction set has the `SBI` and `CBI` instructions, [which set or clear a given bit in an I/O register.](https://onlinedocs.microchip.com/pr/GUID-0B644D8F-67E7-49E6-82C9-1B2B9ABE6A0D-en-US-19/index.html?GUID-DC827DBD-2D0E-4697-83A9-047661370309)

It is not sensible to reproduce this macro-assembler-style C usage in code that is not solely targeted at Atmel microcontrollers.

**Comment by redfast00** [comment]
Fixed in acdacb341df220339ea6a158a7dde298123b3db6
